include/dlaf/matrix/matrix.tpp

```diff
@@ -41,7 +41,7 @@ Matrix<T, D>::Matrix(Distribution distribution, const LayoutInfo& layout) noexce
   DLAF_ASSERT(this->distribution().localSize() == layout.size(),
               "Size of distribution does not match layout size!", distribution.localSize(),
               layout.size());
-  DLAF_ASSERT(this->distribution().blockSize() == layout.blockSize(), distribution.blockSize(),
+  DLAF_ASSERT(this->distribution().baseTileSize() == layout.blockSize(), distribution.baseTileSize(),
```

Comment:
Revert and do a separate PR.

Reply:
This has been reverted, but leaving open for now as a reminder to open a new PR with the changes.

Reply:
https://github.com/eth-cscs/DLA-Future/pull/909